test/jdk/javax/swing/JToolTip/4644444/bug4644444.java

```diff
@@ -21,334 +21,47 @@
-                        "   & see if tooltip works properly everywhere. "
-                        }
-                };
-
-                String[] exceptionsSet =
-                {
-                        "JToolTip is shown partially when placed very close to screen boundaries",
-                };
-
-                Sysout.setInstructionsWithExceptions(instructionsSet,exceptionsSet);
-
-        }
-
-        public void start (){}
-
-        public void destroy(){
-            if(Sysout.failStatus())    {
-                String failMsg = Sysout.getFailureMessages();
-                failMsg = failMsg.replace('\n',' ');
-                throw new RuntimeException(failMsg);
-            }// End destroy
-        }
-}
-
-
-/****************************************************
- Standard Test Machinery
- DO NOT modify anything below -- it's a standard
-  chunk of code whose purpose is to make user
-  interaction uniform, and thereby make it simpler
-  to read and understand someone else's test.
- ****************************************************/
-
-/**
- This is part of the standard test machinery.
- It creates a dialog (with the instructions), and is the interface
-  for sending text messages to the user.
- To print the instructions, send an array of strings to Sysout.createDialog
-  WithInstructions method.  Put one line of instructions per array entry.
- To display a message for the tester to see, simply call Sysout.println
-  with the string to be displayed.
- This mimics System.out.println but works within the test harness as well
-  as standalone.
+ * @test
+ * @bug 4644444 8076246
+ * @summary JToolTip is shown improperly when placed very close to screen boundaries
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual bug4644444
  */
 
-class Sysout
- {
-   private static TestDialog dialog;
-
-   public static void createDialogWithInstructions( String[] instructions )
-    {
-      dialog = new TestDialog( new Frame(), "Instructions" );
-      dialog.printInstructions( instructions );
-      dialog.show();
-      println( "Any messages for the tester will display here." );
-    }
-
-   public static void createDialog( )
-    {
-      dialog = new TestDialog( new Frame(), "Instructions" );
-      String[] defInstr = { "Instructions will appear here. ", "" } ;
-      dialog.printInstructions( defInstr );
-      dialog.show();
-      println( "Any messages for the tester will display here." );
+public class bug4644444 {
+
+    private static final String INSTRUCTIONS = """
+        1. Move the mouse on the button, so that the tooltip is visible.
+        2. Tooltip should get adjusted itself to show its full length of text.
+        3. Similarly, move the frame to different locations of the screen
+            & see if tooltip works properly everywhere.
+        4. Press 'Pass' if tooltip text is fully visible else press 'Fail'. """;
+
+    private static JFrame frame;
+    private static JButton button;
```

Comment:
Both frame and button can be inside `creatUI()`. If you can use `splitUI` instead of `testUI` that would be even better.

Reply:
> Both frame and button can be inside creatUI()

Updated.

> If you can use splitUI instead of testUI that would be even better.

Since the original bug [JDK-4644444](https://bugs.openjdk.org/browse/JDK-4644444) says `JToolTip is shown partially when placed very close to screen boundaries`, it would not be a good idea to use splitUI. User may not be able to move the testUI near screen boundaries.